distributed/protocol/serialize.py

```diff
@@ -78,7 +79,22 @@ def msgpack_decode_default(obj):
     if "__Enum__" in obj:
         mod = importlib.import_module(obj["__module__"])
         enum_type = getattr(mod, obj["__name__"])
-        obj = getattr(enum_type, obj["name"])
+        return getattr(enum_type, obj["name"])
+
+    if "__Set__" in obj:
+        return set(obj["as-list"])
+
+    if "__Layer__" in obj:
+        mod = importlib.import_module(obj["__module__"])
+        layer_type = getattr(mod, obj["__name__"])
```

Comment:
I thin k that we're going to want to be careful around imports for security reasons.  I recommend that we create a whitelist in configuration of trusted libraries (dask, xarray, dask_ml, cudf, dask_cudf, ...).  We would only choose to import modules that begin with these terms. This would be configurable, probably somewhere in the `distributed.scheduler` config namespace